[{
  "comments": [{
    "author": {
      "login": "steveloughran"
    },
    "body": "HADOOP-13372 implies that swift:// throws an UnknownHostException here; best to catch & log too, in case someone adds swift:// to the list of filesystems. ",
    "commit": "e6f77e066c0780352e036aa19173508d02b836ef",
    "createdAt": "2017-05-05T10:57:36Z",
    "diffHunk": "@@ -75,8 +84,15 @@ private[security] class HDFSCredentialProvider extends ServiceCredentialProvider\n     sparkConf.get(PRINCIPAL).flatMap { renewer =>\n       val creds = new Credentials()\n       nnsToAccess(hadoopConf, sparkConf).foreach { dst =>\n-        val dstFs = dst.getFileSystem(hadoopConf)\n-        dstFs.addDelegationTokens(renewer, creds)\n+          try {\n+            val dstFs = dst.getFileSystem(hadoopConf)\n+            dstFs.addDelegationTokens(renewer, creds)\n+          } catch {\n+            case e: StandbyException =>\n+              logWarning(s\"Namenode ${dst} is in state standby\", e)\n+            case e: RemoteException =>\n+              logWarning(s\"Namenode ${dst} is in state standby\", e)\n+          }",
    "line": 43
  }],
  "prId": 17870
}]